### PR TITLE
Containerapp job support keda msi

### PR DESCRIPTION
**Pull request by njuCZ**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by Greedygre** [comment]
Hi @njuCZ 

Please fix the style issue, recently I have fix the others now it should pass.

File: src/containerapp/azext_containerapp/containerapp_job_decorator.py [hunk 1]
```diff
@@ -328,7 +329,7 @@ def set_up_trigger_configurations(self):
                     if self.get_argument_parallelism():
                         event_trigger_config_def["parallelism"] = self.get_argument_parallelism()
                     # Scale
-                    if "scale" in event_trigger_config_def["scale"]:
+                    if "scale" not in event_trigger_config_def:
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/containerapp_job_decorator.py:332]: Why need to change this line?

File: src/containerapp/azext_containerapp/_params.py [hunk 2]
```diff
@@ -305,6 +305,7 @@ def load_arguments(self, _):
         c.argument('max_executions', type=int, help="Maximum number of job executions to run per polling interval.")
         c.argument('polling_interval', type=int, help="Interval to check each event source in seconds. Defaults to 30s.")
         c.argument('scale_rule_type', options_list=['--scale-rule-type', '--srt'], help="The type of the scale rule.")
+        c.argument('scale_rule_identity', options_list=['--scale-rule-identity', '--sri'], help='Resource ID of a managed identity to authenticate with Azure scaler resource(storage account/eventhub or else), or System to use a system-assigned identity.', is_preview=True)
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_params.py:308]: Should add `is_preview=True` as it is preview?

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**